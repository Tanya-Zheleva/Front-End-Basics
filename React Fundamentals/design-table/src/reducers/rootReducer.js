import { SEARCH } from '../actions/actionTypes/SEARCH';
import { RESET } from '../actions/actionTypes/RESET';

const data = {
    '1': {
        key: '1',
        name: 'John Brown',
        age: '32',
        address: 'New York No. 1 Lake Park',
    },
    '2': {
        key: '2',
        name: 'Joe Black',
        age: '42',
        address: 'London No. 1 Lake Park',
    },
    '3': {
        key: '3',
        name: 'Jim Green',
        age: '32',
        address: 'Sidney No. 1 Lake Park',
    },
    '4': {
        key: '4',
        name: 'Jim Red',
        age: '32',
        address: 'London No. 2 Lake Park',
    },
    '5': {
        key: '5',
        name: 'Danny Evans',
        age: '23',
        address: 'Sofia, Mladost',
    },
    '6': {
        key: '6',
        name: 'Harry Potter',
        age: '17',
        address: 'Hogwarts'
    },
    '7': {
        key: '7',
        name: 'Iva Ivanova',
        age: '21',
        address: 'Gabrovo'
    },
    '8': {
        key: '8',
        name: 'Iva Georgieva',
        age: '33',
        address: 'Novi Sad'
    }
};

const columnsConfig = [
    {
        title: 'Name',
        dataIndex: 'name',
        key: 'name',
        width: '30%'
    },
    {
        title: 'Age',
        dataIndex: 'age',
        key: 'age',
        width: '30%'
    },
    {
        title: 'Address',
        dataIndex: 'address',
        key: 'address',
        width: '30%'
    },
];

const dataKeys = Object.keys(data);

const pagination = {
    defaultPageSize: 3,
    showSizeChanger: true,
    pageSizeOptions: ['3', '6', '9'],
};

const initialState = {
    searchText: '',
    searchCol: '',
    gridData: dataKeys,
    gridColumns: columnsConfig,
    data: data,
    pagination: pagination
};

export function rootReducer(state = initialState, action) {
    if (action.type === SEARCH) {
        const { searchCol, searchText } = action.payload;
        let filteredKeys = state.gridData
            .filter(id => state.data[id][searchCol].toLowerCase().indexOf(searchText) !== -1);

        return Object.assign({}, state, {
            searchText: searchText,
            searchCol: searchCol,
            gridData: filteredKeys
        });
    } else if (action.type === RESET) {
        return Object.assign({}, state, {
            searchText: '',
            searchCol: '',
            gridData: dataKeys
        });
    }

    return state;
}