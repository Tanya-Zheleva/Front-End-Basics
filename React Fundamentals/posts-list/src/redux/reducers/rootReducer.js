import SUBMIT_FORM from '../actions/actionTypes/SUBMIT_FORM';
import DELETE_POST from '../actions/actionTypes/DELETE_POST';

const initialState = {
    postTitle: '',
    postContent: '',
    posts: [],
};

export function rootReducer(state = initialState, action) {
    if (action.type === SUBMIT_FORM) {
        const { postTitle, postContent } = action.payload;
        let post = {};
        post[postTitle] = postContent;


        return Object.assign({}, state, {
            postTitle: action.payload.postTitle,
            postContent: action.payload.postContent,
            posts: state.posts.concat(post)
        });
    } else if (action.type === DELETE_POST) {
        console.log(action.payload);
        
        return state;
    }

    return state;
}