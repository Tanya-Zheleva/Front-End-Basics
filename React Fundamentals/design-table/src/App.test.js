import { rootReducer } from './reducers/rootReducer';

let state;
const data = {
  '1': {
    key: '1',
    name: 'John Brown',
    age: '32',
    address: 'New York No. 1 Lake Park',
  }
}

beforeEach(() => {
  state = {
    searchText: '',
    searchCol: '',
    gridData: ['1'],
    data: data,
  }

  return state;
});

describe('rootReducer', () => {
  it('should return inital state', () => {
    expect(
      rootReducer(undefined, {}))
      .toMatchSnapshot();
  });

  it('should handle resset', () => {
    expect(
      rootReducer(undefined, { type: 'RESET' }))
      .toMatchSnapshot();
  });

  it('should handle search', () => {
    expect(
      rootReducer(state, { type: 'SEARCH', payload: {searchText: 'j', searchCol: 'name'} })
    ).toMatchSnapshot();
  })
});
