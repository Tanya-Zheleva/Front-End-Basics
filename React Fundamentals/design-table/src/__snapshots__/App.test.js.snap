// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootReducer should handle resset 1`] = `
Object {
  "data": Object {
    "1": Object {
      "address": "New York No. 1 Lake Park",
      "age": "32",
      "key": "1",
      "name": "John Brown",
    },
    "2": Object {
      "address": "London No. 1 Lake Park",
      "age": "42",
      "key": "2",
      "name": "Joe Black",
    },
    "3": Object {
      "address": "Sidney No. 1 Lake Park",
      "age": "32",
      "key": "3",
      "name": "Jim Green",
    },
    "4": Object {
      "address": "London No. 2 Lake Park",
      "age": "32",
      "key": "4",
      "name": "Jim Red",
    },
    "5": Object {
      "address": "Sofia, Mladost",
      "age": "23",
      "key": "5",
      "name": "Danny Evans",
    },
    "6": Object {
      "address": "Hogwarts",
      "age": "17",
      "key": "6",
      "name": "Harry Potter",
    },
    "7": Object {
      "address": "Gabrovo",
      "age": "21",
      "key": "7",
      "name": "Iva Ivanova",
    },
    "8": Object {
      "address": "Novi Sad",
      "age": "33",
      "key": "8",
      "name": "Iva Georgieva",
    },
  },
  "gridColumns": Array [
    Object {
      "dataIndex": "name",
      "key": "name",
      "title": "Name",
      "width": "30%",
    },
    Object {
      "dataIndex": "age",
      "key": "age",
      "title": "Age",
      "width": "30%",
    },
    Object {
      "dataIndex": "address",
      "key": "address",
      "title": "Address",
      "width": "30%",
    },
  ],
  "gridData": Array [
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
  ],
  "pagination": Object {
    "defaultPageSize": 3,
    "pageSizeOptions": Array [
      "3",
      "6",
      "9",
    ],
    "showSizeChanger": true,
  },
  "searchCol": "",
  "searchText": "",
}
`;

exports[`rootReducer should handle search 1`] = `
Object {
  "data": Object {
    "1": Object {
      "address": "New York No. 1 Lake Park",
      "age": "32",
      "key": "1",
      "name": "John Brown",
    },
  },
  "gridData": Array [
    "1",
  ],
  "searchCol": "name",
  "searchText": "j",
}
`;

exports[`rootReducer should return inital state 1`] = `
Object {
  "data": Object {
    "1": Object {
      "address": "New York No. 1 Lake Park",
      "age": "32",
      "key": "1",
      "name": "John Brown",
    },
    "2": Object {
      "address": "London No. 1 Lake Park",
      "age": "42",
      "key": "2",
      "name": "Joe Black",
    },
    "3": Object {
      "address": "Sidney No. 1 Lake Park",
      "age": "32",
      "key": "3",
      "name": "Jim Green",
    },
    "4": Object {
      "address": "London No. 2 Lake Park",
      "age": "32",
      "key": "4",
      "name": "Jim Red",
    },
    "5": Object {
      "address": "Sofia, Mladost",
      "age": "23",
      "key": "5",
      "name": "Danny Evans",
    },
    "6": Object {
      "address": "Hogwarts",
      "age": "17",
      "key": "6",
      "name": "Harry Potter",
    },
    "7": Object {
      "address": "Gabrovo",
      "age": "21",
      "key": "7",
      "name": "Iva Ivanova",
    },
    "8": Object {
      "address": "Novi Sad",
      "age": "33",
      "key": "8",
      "name": "Iva Georgieva",
    },
  },
  "gridColumns": Array [
    Object {
      "dataIndex": "name",
      "key": "name",
      "title": "Name",
      "width": "30%",
    },
    Object {
      "dataIndex": "age",
      "key": "age",
      "title": "Age",
      "width": "30%",
    },
    Object {
      "dataIndex": "address",
      "key": "address",
      "title": "Address",
      "width": "30%",
    },
  ],
  "gridData": Array [
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
  ],
  "pagination": Object {
    "defaultPageSize": 3,
    "pageSizeOptions": Array [
      "3",
      "6",
      "9",
    ],
    "showSizeChanger": true,
  },
  "searchCol": "",
  "searchText": "",
}
`;
